src/playground.ts

```diff
@@ -0,0 +1,228 @@
+
+            return this.connectionId;
+        }
+
+        const url = new URL(this.tab.url);
+
+        let connectionId = url.searchParams.get(CONNECTION_PARAMETER);
+
+        if (connectionId === null || connectionId === '') {
+            this.logger.debug('No connection ID found in URL');
+
+            connectionId = this.storage.getItem('connectionId');
+
+            this.logger.debug(
+                connectionId !== null
+                    ? 'Previous connection ID found'
+                    : 'No previous connection ID found',
+            );
+
+            return connectionId;
+        }
+
+        this.logger.debug('Connection ID found in URL');
+
+        this.storage.setItem('connectionId', connectionId);
+
+        return connectionId;
+    }
+
+    public disable(): Promise<void> | void {
+        if (this.syncListener !== null) {
+            this.eventSubscriber.removeListener('tokenChanged', this.syncListener);
+            this.tab.removeListener('urlChange', this.syncListener);
+
+            delete this.syncListener;
+        }
+    }
+
+    private syncToken(connectionId: string, cid: string): void {
+        const iframe = document.createElement('iframe');
+        iframe.setAttribute('src', PLAYGROUND_CONNECT_URL);
+        iframe.setAttribute('sandbox', 'allow-scripts allow-same-origin');
+        iframe.style.visibility = 'hidden';
+        iframe.style.opacity = '0';
+        iframe.style.border = '0';
+        iframe.style.width = '0';
+        iframe.style.height = '0';
+
+        const context = this.contextFactory.createContext();
+
+        iframe.onload = (): void => {
+            if (iframe.contentWindow === null) {
+                if (document.body.contains(iframe)) {
+                    document.body.removeChild(iframe);
+                }
+
+                this.logger.warn('Sync handshake failed');
+
+                return;
+            }
+
+            const listener = (event: MessageEvent): void => {
+                if (event.origin !== PLAYGROUND_ORIGIN || event.data !== connectionId) {
+                    return;
+                }
+
+                window.removeEventListener('message', listener);
+
+                if (document.body.contains(iframe)) {
+                    document.body.removeChild(iframe);
+                }
+
+                this.logger.debug('Sync completed');
+            };
+
+            window.addEventListener('message', listener);
+
+            const payload: SyncPayload = {
+                appId: this.appId,
+                connectionId: connectionId,
+                sdkVersion: this.sdkVersion,
+                tabId: this.tab.id,
+                cid: cid,
+                token: this.tokenProvider.getToken()?.toString() ?? null,
+                context: {
+                    ...(context.page && {page: context.page}),
+                    ...(context.campaign && {campaign: context.campaign}),
```

Comment:
These two lines here make any falsy value not add the key to the resulting object `ownProperties`. Is that the intention?

Is there a reason that passing an explicit falsy value (I think null and undefined for these) is unwanted?